### DISCUSSION TITLE
Are all leaderboard results evaluated under 0-shot?

### DISCUSSION THREAD

**Discussion by zhimin-z** [open-compass/opencompass#688] [Q&A] [answered]
https://opencompass.org.cn/leaderboard-llm

**Comment by yingfhu** [comment]
No, the n-shot setting for each dataset is different, you can click the score of the specific dataset on leaderboard to check its configuration. It will tell whether its 0-shot or n-shot from config file.

**Reply by zhimin-z** [reply]
Thanks for your quick reply. I randomly check one config, but I do not know where I should check further to confirm the number of shots. For example, this one: https://github.com/open-compass/opencompass/blob/main/configs/datasets/nq/nq_gen_0356ec.py

**Reply by yingfhu** [reply]
The n-shot might be implicit, you need to read the config especially the content in `retriever` and `prompt_template` filed. In your example, the nq is tested on three settings `0-shot, 1-shot and 5-shot`.

**Reply by yingfhu** [reply]
This one is special, normally we won't use multiple shot settings in one config. The nq score on leaderboard is 0-shot

**Reply by zhimin-z** [reply]
Thanks for the quick replies!

How do you know that this nq is tested on three settings 0-shot, 1-shot, and 5-shot? And how do you know the leaderboards is 0-shot rather than the others?

Due to time constraints, I am a bit lost in the code snippet that I should read...Any trick that can save me time?

Thanks in advance!

**Reply by yingfhu** [reply, marked as answer]
0-shot :
```
retriever=dict(type=ZeroRetriever),
```
k-shot :
```
retriever=dict(type=FixKRetriever, fix_id_list=list(range(k))),
```
use 0,1,5 settings
```
for k in [0, 1, 5]: xxxxx
```

We use 0-shot for nq in leaderboard is not shown, i mentioned that it is a special case, normally we won't use multiple shot settings in one config.
